exercises/dnd-character/canonical-data.json

```diff
@@ -0,0 +1,170 @@
+{
+  "exercise": "dnd-character",
+  "version": "1.0.0",
+  "comments": [
+    "The random generator 'ability' can be property-tested.",
+    "The pure function 'modifier' can be tested totally.",
+    "The random generator 'character' should return some",
+    "kind of structure with six random abilities and a",
+    "hitpoint score. This structure can be property-tested",
+    "in the same way as 'ability' and 'modifier'.",
+    "",
+    "It may be fun to provide a 'name' property for characters",
+    "that may or may not be randomly generated but not tested."
+  ],
+  "cases": [
+    {
+      "description": "random ability is within range",
+      "property": "ability",
+      "input": {},
+      "expected": "score >= 3 && score <= 18"
+    },
+    {
+      "description": "random character is valid",
+      "property": "character",
+      "input": {},
+      "expected": {
+        "strength": "strength >= 3 && strength <= 18",
+        "dexterity": "dexterity >= 3 && dexterity <= 18",
+        "constitution": "constitution >= 3 && constitution <= 18",
+        "intelligence": "intelligence >= 3 && intelligence <= 18",
+        "wisdom": "wisdom >= 3 && wisdom <= 18",
+        "charisma": "charisma >= 3 && charisma <= 18",
+        "hitpoints": "hitpoints == 10 + modifier(constitution)"
+      }
+    },
+    {
+      "description": "the ability modifier",
```

Comment:
This is just a personal preference of mine, but in general I try to keep descriptions as terse as possible. You could consider removing "the" in both this description and in its child case descriptions. The would change a test from `"the ability modifier for the score 3 is -4"` to `"ability modifier for score 3 is -4"`. But feel free to ignore.

Reply:
Thanks.